imap_processing/ultra/l1c/spacecraft_pset.py

```diff
@@ -61,23 +61,31 @@ def calculate_spacecraft_pset(
     # calculate background rates
     background_rates = get_spacecraft_background_rates()
 
-    # TODO: calculate sensitivity and interpolate based on energy.
+    efficiencies = ancillary_files["l1c-90sensor-efficiencies"]
+    geometric_function = ancillary_files["l1c-90sensor-gf"]
+
+    df_efficiencies = pd.read_csv(efficiencies)
+    df_geometric_function = pd.read_csv(geometric_function)
+    sensitivity = interpolate_sensitivity(df_efficiencies, df_geometric_function)
 
     # Calculate exposure
-    constant_exposure = TEST_PATH / "ultra_90_dps_exposure.csv"
+    constant_exposure = ancillary_files["l1c-90sensor-dps-exposure"]
     df_exposure = pd.read_csv(constant_exposure)
     exposure_pointing = get_spacecraft_exposure_times(df_exposure)
 
     # For ISTP, epoch should be the center of the time bin.
     pset_dict["epoch"] = de_dataset.epoch.data[:1].astype(np.int64)
-    pset_dict["counts"] = counts
-    pset_dict["latitude"] = latitude
-    pset_dict["longitude"] = longitude
+    pset_dict["counts"] = counts[..., np.newaxis]
+    pset_dict["latitude"] = latitude[..., np.newaxis]
+    pset_dict["longitude"] = longitude[..., np.newaxis]
     pset_dict["energy_bin_geometric_mean"] = energy_bin_geometric_means
-    pset_dict["background_rates"] = background_rates
-    pset_dict["exposure_factor"] = exposure_pointing
+    pset_dict["background_rates"] = background_rates[..., np.newaxis]
+    pset_dict["exposure_factor"] = exposure_pointing.to_numpy()[..., np.newaxis]
```

Comment:
Ah, here's where the epoch dim is getting added at the end; I _think_ you'll just need to change around the ... and newaxis.